app/Storage/Redis/RequestStore.php

```diff
@@ -47,27 +47,44 @@ public function find(Token $token, $requestId)
      * @param Token $token
      * @param int $page
      * @param int $perPage
+     * @param string $sort
      * @return Collection|static
      */
-    public function all(Token $token, $page = 1, $perPage = 50)
+    public function all(Token $token, $page = 1, $perPage = 50, $sorting = 'oldest')
     {
-        return collect(
+        $requests = collect(
             $this->redis->hgetall(Request::getIdentifier($token->uuid))
         )
         ->filter()
         ->map(
             function ($request) {
                 return json_decode($request);
             }
-        )->sortBy(
-            function ($request) {
-                return Carbon::createFromFormat(
-                    'Y-m-d H:i:s',
-                    $request->created_at
-                )->getTimestamp();
-            },
-            SORT_DESC
-        )->forPage(
+        );
+        
+        if ($sorting === 'newest') {
+            $requests = $requests->sortByDesc(
+                function ($request) {
+                    return Carbon::createFromFormat(
+                        'Y-m-d H:i:s',
+                        $request->created_at
+                    )->getTimestamp();
+                },
+                SORT_DESC
+            );
+        } else {
+            $requests = $requests->sortBy(
+                function ($request) {
+                    return Carbon::createFromFormat(
+                        'Y-m-d H:i:s',
+                        $request->created_at
+                    )->getTimestamp();
+                },
+                SORT_DESC
+            );
+        }
```

Comment:
I'd do it like this, and wouldn't they both sort the same anyways?

```suggestion
            $requests = $requests->sortBy(
                function ($request) {
                    return Carbon::createFromFormat(
                        'Y-m-d H:i:s',
                        $request->created_at
                    )->getTimestamp();
                },
                $sorting === 'newest' ? SORT_DESC : SORT_ASC
            );
```

Reply:
Hey, thanks. That is most likely the better option. I am no PHP expert, so I am not too familiar with the sorting syntax